pkg/operation/botanist/cleaner.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+	"github.com/gardener/gardener/pkg/utils"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+// DeleteAll calls delete for all objects in the given list.
+//
+// Not found errors are being ignored.
+func DeleteAll(ctx context.Context, c client.Client, list runtime.Object, opts ...client.DeleteOptionFunc) error {
+	return meta.EachListItem(list, func(obj runtime.Object) error {
+		if err := c.Delete(ctx, obj, opts...); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
WDYT about executing this in parallel?

Considering a deletion with options `ForegroundDeletion` as well as `GracePeriods` may take an increased amount of time for a single resource compared to the overall time this function is allowed to run (see https://github.com/gardener/gardener/blob/e63ff1627689e560193e4925dd75902467a025bb/pkg/operation/botanist/cleanup.go#L174).

Reply:
Good idea! Also did the same for `FinalizeAll`. In the future, this will hopefully anyways be gone if we can do a single call with `DeleteCollection` once it's there.